Docker/README.md

```diff
@@ -0,0 +1,278 @@
+# Docker 101 #
+
+## What is Docker? ##
+
+Allows the creation of light-weight containers that run within a host in a manner avoiding the overhead of a full-fledged virtual machine. The docker image specifies a base image (via the Dockerfile FROM option) to build upon along with a definition specifying the dependencies of a given application one is containerizing and a packaging mechanism to share the Docker images.
+
+
+## What is Docker Compose? ##
+
+Allows defining and running a multi-container Docker application. A docker-compose.yml defines the services pulled into the application (e.g., Docker images to include in the application).  
+
+
+## How is Docker being used? ##
+
+* lightweight container for development/testing/debugging in a defined and easily repeatable environment in a team environment
+
+
+## Requirements ##
+
+* [Docker](https://docs.docker.com/engine/installation/)
+* [Docker Compose](https://docs.docker.com/compose/install/)  
+  * version 1.13 or higher as of 2017-05-10
+
+
+## Docker: Usage ##
+
+First usage steps:
+
+1. Acquire the Docker image. There are two options:
+    1. Download the prebuilt images from [ualibraries DockerHub](https://hub.docker.com/r/ualibraries/)
+        * `docker pull ualibraries/${image_name}:${option_tag}`
+    2. Build from `Dockerfile` definition 
+        * `docker build -t ualibraries/${image_name}:${option_tag} -f ${dockerfile} ${dockerfile_directory`
+
+2. Run the Docker image creating a container
+
+    * `docker run -d [OPTIONS] IMAGE[:TAG|@DIGEST] [COMMAND] [ARG...]`
+
+After initial `docker run`, 
+
+* Exec shell within container
+  * `docker exec -it ${container_id}  bash`
+
+* `docker stop ${container_id}`
+
+* `docker start ${container_id}`
+
+
+
+## Docker-Compose: Usage  ##
+
+* Pull images associated with a service defined in a docker-compose.yml
+  * `docker-compose -f docker-compose.yml pull`
+* Setup application
+  * `docker-compose -f docker-compose.yml up -d`
+* Subsequent uses
+  * `docker-compose -f docker-compose.yml stop`
+  * `docker-compose -f docker-compose.yml start`
+  * `docker-compose -f docker-compose.yml restart`
+
+
+## Maintenance/Creation Tasks ##
+
+### Creating from scratch  ###
+
+* Create a GitHub repo to contain the files
+* Start with [README.md template](#readme-template)
+* Populate the supporting Docker files (e.g, Dockerfile, docker-compose.yml)
+* Build the Docker image from `Dockerfile` definition
+* Test image
+* Push to https://hub.docker.com/r/ualibraries/
```

Comment:
Side point is how does someone get access to this ualibraries dockerhub?  To push/update images to it? Do we have to go bug Sam or Alex? etc.